### PR TITLE
Turn off Analytics opt-in notification when no analytics id is set.

### PR DESCRIPTION
**Pull request by patflynn**
I had to make a bunch of modifications to get this to work and tests to pass.

### CODE REVIEW

**Comment by patflynn** [comment]
@loosebazooka @nbashirbello @elharo

File: common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import com.google.common.annotations.VisibleForTesting;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * Use an instance of this class to read flags provided to your plugin through an {@code
```
Comment (elharo) [inline common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java]: nit an --> a

File: common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ * config.properties} file placed in your plugin's resource root directory.
+ */
+public class PropertiesFileFlagReader implements FlagReader {
+
+  private static final String DEFAULT_PROPERTIES_FILE_PATH = "/config.properties";
+  private final Properties properties;
```
Comment (elharo) [inline common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java]: Might just initialize here

File: common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  public PropertiesFileFlagReader() {
+    this(DEFAULT_PROPERTIES_FILE_PATH);
+  }
+
+  @VisibleForTesting
+  protected PropertiesFileFlagReader(String propertiesFilePath) {
+    properties = new Properties();
+    InputStream resourceAsStream = getClass().getClassLoader()
```
Comment (elharo) [inline common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java]: you need to close this stream in a finally block if it isn't null

File: common-lib/src/test/java/com/google/gct/idea/flags/PropertiesFileFlagReaderTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test cases for {@link PropertiesFileFlagReader}.
+ */
+public class PropertiesFileFlagReaderTest {
+
+  @Test
+  public void testGetFlagString_flatExists() throws Exception {
```
Comment (elharo) [inline common-lib/src/test/java/com/google/gct/idea/flags/PropertiesFileFlagReaderTest.java]: probably don't need these throws Exception

File: google-login-plugin/src/com/google/gct/login/stats/UsageTrackerManager.java [hunk 5]
```diff
@@ -1,78 +1,84 @@
...
-        } catch (SecurityException ex) {
-            LOG.error(ex.getMessage());
-            return null;
-        } catch (IOException ex) {
-            LOG.error(ex.getMessage());
-            return null;
-        }
+  /**
+   * @return true if running on IntelliJ platform and the usage tracking property exists; false
+   * otherwise
```
Comment (elharo) [inline google-login-plugin/src/com/google/gct/login/stats/UsageTrackerManager.java]: indent or move to one line

File: google-login-plugin/testSrc/com/google/gct/login/stats/UsageTrackerManagerTest.java [hunk 6]
```diff
@@ -1,48 +1,65 @@
...
-        UsageTrackerManager usageTrackerManager = new UsageTrackerManager(mockComponent);
-        assertNotNull(usageTrackerManager.getAnalyticsProperty());
-    }
+
+  @Mock
+  private PropertiesComponent mockComponent;
+  @Mock
+  private PluginFlags mockFlags;
+
+  UsageTrackerManager manager;
```
Comment (elharo) [inline google-login-plugin/testSrc/com/google/gct/login/stats/UsageTrackerManagerTest.java]: private

File: common-lib/src/test/java/com/google/gct/idea/flags/PropertiesFileFlagReaderTest.java (hunk 4 above)
Comment (patflynn) [inline common-lib/src/test/java/com/google/gct/idea/flags/PropertiesFileFlagReaderTest.java]: intellij seems to generate test methods with exception for some reason.

File: common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+  private final Properties properties;
+
+  public PropertiesFileFlagReader() {
+    this(DEFAULT_PROPERTIES_FILE_PATH);
+  }
+
+  @VisibleForTesting
+  protected PropertiesFileFlagReader(String propertiesFilePath) {
+    properties = new Properties();
+    InputStream resourceAsStream = null;
```
Comment (elharo) [inline common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java]: could simply name this in; up to you

File: common-lib/src/test/resources/com/google/gct/idea/flags/flags_test.properties [hunk 8]
```diff
@@ -0,0 +1,2 @@
+test.flag=testing123
+test.empty.flag=
```
Comment (elharo) [inline common-lib/src/test/resources/com/google/gct/idea/flags/flags_test.properties]: might want to put a \n at eof here

File: common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+    try {
+      resourceAsStream = getClass().getClassLoader()
+          .getResourceAsStream(propertiesFilePath);
+      if (resourceAsStream == null) {
+        throw new IllegalArgumentException("Failed to find the plugins property configuration file"
```
Comment (elharo) [inline common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java]: not an argument problem; IllegalStateException might fit better

**Comment by elharo** [comment]
lgtm

File: google-login-plugin/src/com/google/gct/login/stats/UsageTrackerService.java [hunk 10]
```diff
@@ -48,11 +49,12 @@ public static UsageTracker getInstance() {
    *
    */
   static UsageTracker getInstance(@Nullable String key) {
-    UsageTracker instance = (key == null) ? null : COLLECTOR.findSingle(key);
-    if (instance != null) {
-      return instance;
+    if (usageTrackerManager.isTrackingEnabled()) {
```
Comment (loosebazooka) [inline google-login-plugin/src/com/google/gct/login/stats/UsageTrackerService.java]: There is an issue with putting this check here, enabling tracking via dialog after a tracking event happens doesn't change the tracker to the google tracker.  They have to enable tracking, then restart the IDE.  I believe the intellij/studio guys do this check in their tracker.

File: google-login-plugin/testSrc/com/google/gct/login/stats/UsageTrackerManagerTest.java [hunk 11]
```diff
@@ -1,48 +1,65 @@
 package com.google.gct.login.stats;
 
+import static org.mockito.Mockito.when;
+
```
Comment (loosebazooka) [inline google-login-plugin/testSrc/com/google/gct/login/stats/UsageTrackerManagerTest.java]: ? there's a lot of spaces in this import list? Did intellij autoformat this?

Comment (patflynn) [inline google-login-plugin/testSrc/com/google/gct/login/stats/UsageTrackerManagerTest.java]: yep!

File: common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java (hunk 7 above)
Comment (patflynn) [inline common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java]: done. shorter is better.

File: common-lib/src/test/resources/com/google/gct/idea/flags/flags_test.properties (hunk 8 above)
Comment (patflynn) [inline common-lib/src/test/resources/com/google/gct/idea/flags/flags_test.properties]: done.

File: common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java (hunk 9 above)
Comment (patflynn) [inline common-lib/src/main/java/com/google/gct/idea/flags/PropertiesFileFlagReader.java]: eeeh. It kind of is caused by a bad file path..whatever.

File: google-login-plugin/src/com/google/gct/login/stats/UsageTrackerService.java (hunk 10 above)
Comment (patflynn) [inline google-login-plugin/src/com/google/gct/login/stats/UsageTrackerService.java]: good point. fixed.

**Pull request merged**